Mathlib/Data/Finset/Insert.lean

```diff
@@ -219,9 +219,14 @@ instance (i : α) : Unique ({i} : Finset α) where
 @[simp]
 lemma default_singleton (i : α) : ((default : ({i} : Finset α)) : α) = i := rfl
 
-instance Nontrivial.instDecidablePred [DecidableEq α] :
-    DecidablePred (Finset.Nontrivial (α := α)) :=
-  inferInstanceAs (DecidablePred fun s ↦ ∃ a ∈ s, ∃ b ∈ s, a ≠ b)
+instance Nontrivial.instDecidablePred : DecidablePred (Finset.Nontrivial (α := α)) := fun s =>
+  Quotient.recOnSubsingleton (motive := fun (s : Multiset α) =>
+      (h : s.Nodup) → Decidable (Finset.Nontrivial ⟨s, h⟩))
+    s.val (fun l h => match l with
+      | [] => isFalse (by simp)
+      | [_] => isFalse (by simp [Finset.toSet])
+      | a :: b :: _ => isTrue ⟨a, by simp, b, by simp,
+        List.ne_of_not_mem_cons (List.nodup_cons.mp h).left⟩) s.nodup
```

Comment:
Shorter as:
```lean-4
instance Nontrivial.instDecidablePred : DecidablePred (Finset.Nontrivial (α := α)) :=
  fun _ => decidable_of_iff _ <| Finset.one_lt_card_iff_nontrivial
```
though maybe your spelling has other advantages. If you decide to keep it, can you add a comment mentioning why you avoid a shorter construction via `one_lt_card_iff_nontrivial`?

bors d+